views/actions/packages.js

```diff
@@ -65,26 +80,32 @@ export function getSubscribedPackages() {
   };
 }
 
-function createAddPackageRequest(trackingNumber) {
+function createAddPackageRequest(carrier, trackingNumber) {
   return {
     path: '/packages',
     method: 'POST',
     body: {
       package: {
+        carrier,
         trackingNumber,
       },
     },
   };
 }
 
-export function requestToAddPackage(trackingNumber) {
-  const request = createAddPackageRequest(trackingNumber);
+export function requestToAddPackage(carrier, trackingNumber) {
+  const request = createAddPackageRequest(carrier, trackingNumber);
   return (dispatch, getState) => {
+    query(request).then((response) => {
+      response.json().then((json) => {
+        if (!response.ok) {
+          dispatch(addAddPackageError());
```

Comment:
I would consider doing something like `if (!response.ok) throw Error("Sorry, your package was not found.")`
Then in the catch block `dispatch(addAddPackageError(error.message))`